### PR TITLE
feat(query): moved 'Invalid Schema External Documentation URL' to general

### PR DESCRIPTION
**Pull request by rafaela-soares**
Signed-off-by: Rafaela Soares <rafaela.soares@checkmarx.com>

**Proposed Changes**
- Moved "Invalid Schema External Documentation URL" query to general
- Implementation of `walk` function

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: assets/queries/openAPI/general/invalid_schema_external_documentation_url/query.rego [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+	result := {
+		"documentId": doc.id,
+		"searchKey": sprintf("definitions.{{%s}}.externalDocs.url", [s]),
+		"issueType": "IncorrectValue",
+		"keyExpectedValue": "Schema External Documentation URL is a valid URL",
+		"keyActualValue": "Schema External Documentation URL is not a valid URL",
+		"overrideKey": version,
+	}
+}
```
Comment (felipe-avelar) [inline assets/queries/openAPI/general/invalid_schema_external_documentation_url/query.rego:57]: I think this to rules can be merged as following:
```suggestion
CxPolicy[result] {
	doc := input.document[i]
	version := openapi_lib.check_openapi(doc)
	def := get_schema_definitions(doc, version)

	url := def[s].externalDocs.url
	not openapi_lib.is_valid_url(url)

	result := {
		"documentId": doc.id,
		"searchKey": sprintf("%s.{{%s}}.externalDocs.url", [get_schema_definitions_path(version), s]),
		"issueType": "IncorrectValue",
		"keyExpectedValue": "Schema External Documentation URL is a valid URL",
		"keyActualValue": "Schema External Documentation URL is not a valid URL",
		"overrideKey": version,
	}
}

get_schema_definitions(docs, version) = obj {
  version == "3.0"
  obj := doc.components.schemas
} else = obj {
  version == "2.0"
  obj := doc.definitions
}

get_schema_definitions_path(version) = ret {
  version == "3.0"
  ret := "components.schemas"
} else = obj {
  version == "2.0"
  ret := "definitions"
}
```

**Review by felipe-avelar — CHANGES_REQUESTED** [review]
Please, check comments

File: assets/queries/openAPI/general/invalid_schema_external_documentation_url/query.rego (hunk 1 above)
Comment (rafaela-soares) [reply assets/queries/openAPI/general/invalid_schema_external_documentation_url/query.rego]: A nice suggestion 🚀 Thank you! We can merge `get_schema_definitions` and `get_schema_definitions_path`:

```
get_schema_info(doc, version) = schemaInfo {
	version == "3.0"
	schemaInfo := {"obj": doc.components.schemas, "path": "components.schemas"}
} else = schemaInfo {
	version == "2.0"
	schemaInfo := {"obj": doc.definitions, "path": "definitions"}
}
```

**Review by felipe-avelar — APPROVED** [review]
LGTM

**Pull request merged**